specification/compute/resource-manager/Microsoft.ContainerInstance/2017-08-01-preview/containerInstance.json

```diff
@@ -0,0 +1,819 @@
+                  "type": "array",
+                  "items": {
+                    "$ref": "#/definitions/Container"
+                  }
+                },
+                "imageRegistryCredentials": {
+                  "type": "array",
+                  "items": {
+                    "$ref": "#/definitions/ImageRegistryCredential"
+                  }
+                },
+                "restartPolicy": {
+                  "type": "string",
+                  "description": "- `always` Always restart\n",
+                  "enum": [
+                    "always"
+                  ],
+                  "x-ms-enum": {
+                    "name": "ContainerRestartPolicy",
+                    "modelAsString": false
+                  }
+                },
+                "ipAddress": {
+                  "$ref": "#/definitions/IpAddress"
+                },
+                "osType": {
+                  "type": "string",
+                  "description": "The base level OS type required by the containers in the group.",
+                  "enum": [
+                    "Windows",
+                    "Linux"
+                  ],
+                  "x-ms-enum": {
+                    "name": "OperatingSystemTypes",
+                    "modelAsString": false
+                  }
+                },
+                "state": {
+                  "readOnly": true,
+                  "type": "string",
+                  "description": "The state of the container group. Only valid in response."
+                },
+                "volumes": {
+                  "type": "array",
+                  "items": {
+                    "$ref": "#/definitions/Volume"
+                  }
+                }
+              },
+              "required": [
+                "containers",
+                "osType"
+              ]
+            }
+          }
+        }
+      ]
+    },
+    "ImageRegistryCredential": {
+      "description": "Image registry credential.",
+      "type": "object",
+      "properties": {
+        "server": {
+          "type": "string",
+          "description": "docker image registry server, without protocol such as \"http\" and \"https\"."
+        },
+        "username": {
+          "type": "string"
+        },
+        "password": {
+          "type": "string",
+          "description": "Password for the private registry. In most cases, this will be pulled from KeyVault."
+        }
+      },
+      "required": [
+        "server",
+        "username"
+      ]
+    },
+    "IpAddress": {
+      "description": "IP address for the group.",
+      "type": "object",
+      "properties": {
+        "ports": {
+          "type": "array",
+          "items": {
+            "$ref": "#/definitions/Port"
+          }
+        },
+        "type": {
+          "type": "string",
+          "description": "Determines whether the IP is exposed to the public internet.",
+          "enum": [
+            "Public"
```

Comment:
No, public IP is all that's supported currently. Eventually, we will add private IPs on a customer vnet.